trieste/acquisition/rule.py

```diff
@@ -200,14 +202,144 @@ def acquire(
                 self._num_query_points - 1
             ):  # greedily allocate remaining batch elements
                 self._acquisition_function = self._builder.update_acquisition_function(
-                    self._acquisition_function, datasets, models, pending_points=points
+                    self._acquisition_function,
+                    datasets,
+                    models,
+                    pending_points=points,
+                    new_optimization_step=False,
                 )
                 chosen_point = self._optimizer(search_space, self._acquisition_function)
                 points = tf.concat([points, chosen_point], axis=0)
 
         return points
 
 
+class AsyncEGO(AcquisitionRule[State[Optional["AsyncEGO.State"], TensorType], SP_contra]):
+    """AsyncEGO rule, as name suggests, is designed for async BO scenarios.
+    By async BO we understand a use case when multiple objective function can be launched in parallel
+    and are expected to arrive at different times. Instead of waiting for the rest of observations to return,
+    we want to immediately use acquisition function to launch a new observation and avoid wasting computational resources.
+
+    To make the best decision about next point to observe, acquisition function needs to be aware of currently running observations.
+    We call such points "pending". AsyncEGO provides a ``AsyncEGO.State`` object that keeps track of pending points
```

Comment:
I think we should differentiate between "pending" points used by greedy acquisition functions (points that have been selected but haven't been executed yet) and "outstanding" points, which have been executed but haven't returned.

```suggestion
    We call such points "outstanding". AsyncEGO provides a ``AsyncEGO.State`` object that keeps track of outstanding points
```

Reply:
Can you elaborate a bit what's the difference between pending and outstanding? I don't think I understand.

Once the point was chosen by the acquisition function, there is no difference any more, as I understand it. It could have been chosen 5 calls ago or just now, it is still treated exactly the same - as a point which was selected for observations, but result for which hasn't arrived yet.

Reply:
It's like the difference between things in your Amazon basket, and things that you've ordered but haven't arrived yet.

* Greedy acquistiion functions choose points one at a time. The first points are pending until we've selected the entire batch.
* Asyn BO choose some points (greedily) and then ask for them to be evaluated (like normal BO), but then starts another loop before all of them have finished executing. The ones that haven't finished yet are outstanding.